### PR TITLE
Add example for mapping role names between Keycloak and Spring Boot

### PR DESCRIPTION
**Pull request by little-pinecone**
The reason for this PR is the ambiguity regarding the setting of the role name prefix. I've seen many Keycloak-Spring Boot tutorials where the `ROLE_` prefix was explicitly set which was redundant as the `SimpleAuthorityMapper` defaults to the correct prefix.

Since Spring's documentation for this mapper is very clear, I suspect that unnecessary configuration may come from a misunderstanding of the _"Use (…) `SimpleAuthorityMapper` to insert the `ROLE_` prefix"_. I think the code example I suggest can reduce superfluous configuration in many projects that rely on this mapper.

Furthermore, the example shows how to convert role names to upper case.

### CODE REVIEW

File: securing_apps/topics/oidc/java/spring-security-adapter.adoc [hunk 1]
```diff
@@ -193,8 +193,32 @@ Spring Security, when using role-based authentication, requires that role names
 For example, an administrator role must be declared in Keycloak as `ROLE_ADMIN` or similar, not simply `ADMIN`.
 
 The class `org.keycloak.adapters.springsecurity.authentication.KeycloakAuthenticationProvider`            supports an optional `org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper`            which can be used to map roles coming from Keycloak to roles recognized by Spring Security.
-Use, for example, `org.springframework.security.core.authority.mapping.SimpleAuthorityMapper` to insert the `ROLE_` prefix and convert the role name to upper case.
-The class is part of Spring Security Core module.
+Use, for example, `org.springframework.security.core.authority.mapping.SimpleAuthorityMapper` which allows for case conversion and the addition of a prefix (which defaults to `ROLE_`).
```
Comment (andymunro) [inline securing_apps/topics/oidc/java/spring-security-adapter.adoc]: @little-pinecone Add a comma before the first which.

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks, @little-pinecone.  Just one change requested.

**Comment by andymunro** [comment]
To resolve the Expected message, use this command:  git pull upstream main

**Comment by ahus1** [comment]
@little-pinecone - thanks for sharing this documentation update as a pull request and sorry for us taking so long to review it. 

To you happens to have an example project at hand which implements it this way so I can try out this or a similar example?

**Comment by little-pinecone** [comment]
@ahus1 The example project can be found here: https://github.com/little-pinecone/keycloak-spring-boot/blob/master/src/main/java/in/keepgrowing/keycloakspringboot/security/config/SecurityConfig.java

Running the app is described in the project [README](https://github.com/little-pinecone/KEYCLOAK-SPRING-BOOT).

**Comment by ahus1** [comment]
The contents of this PR have been moved to the main repository as the documentation has been moved there: https://github.com/keycloak/keycloak/pull/19536

**Pull request closed without merging**